ketcher-autotests/tests/Structure-Creating-&-Editing/Actions-With-Structures/Selection-Tool/Lasso-Selection-Tool/lasso-tool.spec.ts

```diff
@@ -0,0 +1,258 @@
+  });
+
+  test('Drag atom/bond/molecule', async ({ page }) => {
+    /**
+     * Test case: EPMLSOPKET-1339
+     * Description: Atom/bond/molecule is moved to another place. Structure is not changed. Only selected part changed their place.
+     */
+    await openFileAndAddToCanvas('KET/two-benzene-with-atoms.ket', page);
+    await selectLasso(page);
+    const point = await selectObjects(page, 100, 100);
+    await clickOnAtom(page, 'C', 5);
+    await dragMouseTo(point.x + xDelta, point.y - yDelta, page);
+  });
+
+  test('Select the reaction components', async ({ page }) => {
+    /**
+     * Test case: EPMLSOPKET-1340
+     * Description: Hover and selection of plus sign and reaction arrow
+     */
+    await openFileAndAddToCanvas('Rxn-V2000/reaction_4.rxn', page);
+    await selectLasso(page);
+    const point = await getCoordinatesOfTheMiddleOfTheScreen(page);
+    await page.mouse.move(point.x - 270, point.y + 10);
+    await page.mouse.down();
+    await page.mouse.up();
+    await clickCanvas(page);
+
+    await page.mouse.move(point.x, point.y + 5);
+    await page.mouse.down();
+    await page.mouse.up();
+    await clickCanvas(page);
+
+    await page.keyboard.down('Shift');
+    await page.mouse.click(point.x - 270, point.y + 10);
+    await page.mouse.click(point.x, point.y + 5);
+    await page.keyboard.up('Shift');
+    await clickCanvas(page);
+
+    await page.keyboard.press(`${modifier}+KeyA`);
+  });
+
+  test('Drag the reaction components', async ({ page }) => {
+    /**
+     * Test case: EPMLSOPKET-1342
+     * Description: Selected structures and components are moved to the another place.
+     */
+    await openFileAndAddToCanvas('Rxn-V2000/reaction_4.rxn', page);
+    await selectLasso(page);
+    const point = await selectObjects(page, 300, 200);
+    await clickOnAtom(page, 'C', 10);
+    await dragMouseTo(point.x - 100, point.y - 200, page);
+  });
+
+  test('Fuse atoms together', async ({ page }) => {
+    /**
+     * Test case: EPMLSOPKET-1343
+     * Description: Atoms are fused.
+     */
+    await openFileAndAddToCanvas('KET/two-benzene-with-atoms.ket', page);
+    await selectLasso(page);
+    await clickOnAtom(page, 'C', 4);
+    const atomPoint = await getAtomByIndex(page, { label: 'C' }, 7);
+    await dragMouseTo(atomPoint.x, atomPoint.y, page);
+  });
+
+  test('Fuse bonds together', async ({ page }) => {
+    /**
+     * Test case: EPMLSOPKET-2861
+     * Description: When benzene is merged with stereo bond, benzene's bond is changed.
+     * When stereo bond is merged with benzene, benzene's bond is not changed.
+     * No new labels (abs, Chiral) appears.
+     */
+    await drawBenzeneRing(page);
+    await selectNestedTool(page, BondTool.SINGLE_AROMATIC);
+    const coordinates = await getCoordinatesTopAtomOfBenzeneRing(page);
+    await page.mouse.click(coordinates.x + xDelta, coordinates.y - yDelta);
+    await page.getByTestId('select-rectangle').click();
+    delay(DELAY_IN_SECONDS.TWO);
```

Comment:
Why do we need delay? Can we remove it?

Reply:
delay is to avoid douleClick event here on the selection-rectangle

Reply:
Wouldn't `await selectNestedTool(page, SelectTool.LASSO_SELECTION);` help?

Reply:
selectNestedTool only works if we already selected the tool and wanted to switch the selected tools, while in this situation the selection tool is not selected after the paste BenzeneRing in the editor